src/main/java/org/folio/rest/impl/HoldingsRecordsSourceAPI.java

```diff
@@ -0,0 +1,193 @@
+import org.folio.rest.persist.PgExceptionUtil;
+import org.folio.rest.persist.PgUtil;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.persist.Criteria.Limit;
+import org.folio.rest.persist.Criteria.Offset;
+import org.folio.rest.persist.cql.CQLWrapper;
+import org.folio.rest.tools.messages.MessageConsts;
+import org.folio.rest.tools.messages.Messages;
+import org.folio.rest.tools.utils.TenantTool;
+import org.z3950.zing.cql.CQLParseException;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+public class HoldingsRecordsSourceAPI implements org.folio.rest.jaxrs.resource.HoldingsRecordsSources {
+
+  private static final String REFERENCE_TABLE = "holdings_records_source";
+  private static final String LOCATION_PREFIX = "/holdings-records-sources/";
+  private static final Logger log = LoggerFactory.getLogger(HoldingsRecordsSourceAPI.class);
+  private final Messages messages = Messages.getInstance();
+
+  @Override
+  public void getHoldingsRecordsSources(String query, int offset, int limit, String lang,
+    Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+      vertxContext.runOnContext(v -> {
+        try {
+          String tenantId = TenantTool.tenantId(okapiHeaders);
+          CQLWrapper cql = getCQL(query, limit, offset);
+          PostgresClient.getInstance(vertxContext.owner(), tenantId).get(REFERENCE_TABLE,
+            HoldingsRecordsSource.class,new String[]{"*"}, cql, true, true,
+            reply -> {
+              if (reply.succeeded()) {
+                HoldingsRecordsSources sources = new HoldingsRecordsSources();
+                List<HoldingsRecordsSource> source = reply.result().getResults();
+                sources.setHoldingsRecordsSources(source);
+                sources.setTotalRecords(reply.result().getResultInfo().getTotalRecords());
+                asyncResultHandler.handle(succeededFuture(GetHoldingsRecordsSourcesResponse
+                  .respond200WithApplicationJson(sources)));
+              } else {
+                log.error(reply.cause().getMessage(), reply.cause());
+                asyncResultHandler.handle(succeededFuture(GetHoldingsRecordsSourcesResponse
+                  .respond400WithTextPlain(reply.cause().getMessage())));
+              }
+            });
+        } catch (Exception e) {
+          log.error(e.getMessage(), e);
+          String message = messages.getMessage(lang, MessageConsts.InternalServerError);
+          if (e.getCause() instanceof CQLParseException) {
+            message = " CQL parse error " + e.getLocalizedMessage();
+            asyncResultHandler.handle(succeededFuture(GetHoldingsRecordsSourcesResponse
+              .respond500WithTextPlain(message)));
+          }
+        }
+      });
```

Comment:
Is there any reason why standard `org.folio.rest.persist.PgUtil#get(...)` method can not be used?